src/main/java/Duke/Storage/TaskStorage.java

```diff
@@ -0,0 +1,166 @@
+    private final String path;
+
+    /**
+     * Constructs a new TaskStorage with the specified path.
+     * Creates the directories and file if they don't already exist.
+     *
+     * @param path The path to the file storing the tasks.
+     * @throws IOException If an IOException is thrown when creating the directories or file.
+     */
+    public TaskStorage(String path) throws IOException {
+        this.path = path;
+        Files.createDirectories(Paths.get(path).getParent());
+        createFileIfNotExist(path);
+    }
+
+    /**
+     * Returns the list of tasks stored on the file.
+     *
+     * @return The list of tasks stored.
+     * @throws IOException If an IOException is thrown when creating or reading from the file.
+     */
+    public List<Task> load() throws IOException {
+        List<Task> list = new ArrayList<>();
+        File file = new File(this.path);
+        Scanner scanner;
+        try {
+            scanner = new Scanner(file);
+        } catch (FileNotFoundException e) {
+            createFileIfNotExist(this.path);
+            return list;
+        }
+
+        while (scanner.hasNext()) {
+            String data = scanner.nextLine();
+            list.add(parseTask(data));
+        }
+
+        return list;
+    }
+
+    /**
+     * Saves the specified list of tasks to the file, overwriting the previous contents on the file.
+     *
+     * @param list The list of elements to save.
+     * @throws IOException If an IOException is thrown when writing to the file.
+     */
+    public void save(List<Task> list) throws IOException {
+        FileWriter writer = new FileWriter(this.path);
+        StringBuilder builder = new StringBuilder();
+        for (Task task : list) {
+            builder.append(encodeTask(task));
+            builder.append("\n");
+        }
+        writer.write(builder.toString());
+        writer.close();
+    }
+
+    /**
+     * Creates the specified file if it does not exist.
+     *
+     * @param path The file to create.
+     * @throws IOException If an IOException is thrown when creating the file.
+     */
+    private static void createFileIfNotExist(String path) throws IOException {
+        File file = new File(path);
+        file.createNewFile();
+    }
+
+    /**
+     * Parses the string argument as a Task.
+     * The string is expected to be in the format "[T|D|E],[0|1],<description>,<date>",
+     * where `[T|D|E]` represents the type of task and `[0|1]` represents the completion status.
+     *
+     * @param data The string to be parsed.
+     * @return The Task object.
+     * @throws FileFormatException If the string is not in the specified format.
+     */
+    private static Task parseTask(String data) throws FileFormatException {
+        String[] splitData = data.split(",", -1);
+        if (splitData.length != 4) {
+            throw new FileFormatException("Expected 4 values, got " + splitData.length);
+        }
+
+        int completionStatus;
```

Comment:
```suggestion
        int completionStatus = 0;
```
The coding standards mentioned that variables should be initialized where they are declared so maybe you could do this?